### DISCUSSION TITLE
How to "read" a converted column in a Dao?

### DISCUSSION THREAD

**Discussion by Adam-Langley** [simolus3/drift#2619] [Q&A]
I have a textcolumn which maps to a custom 'Guid' type using a Dart converter.
When using table reads, this works well - no custom handling of the guid column code necessary.
However, when using a DAO - I cannot find a simple method which will use the converter when 'read'-ing a column that uses a converter.

I have resorted to this...
```
return query.map((e) => CustomModel(mytable.mycolumn.converter.fromSql(e.read(mytable.mycolumn)).get();
```

it's just a little yuck... of course, I can write an extension that wraps this up - just thought - surely there is one somewhere already in the drift code that I've overlooked?

**Comment by simolus3** [comment]
Just to clarify, the problem is using `selectOnly` (or reading the single column via `.read`), right? Doing a full select across all columns should also work in a DAO.

There isn't a solution for this in drift yet - I've added one in 0ed7358f62fce313bd5c77e49dc6717b43330382 which will be part of the next drift release. Until then, can retrofit this to make it a bit easier:

```dart
extension ReadWithConverter on TypedResult {
  D? readWithConverter<D, S extends Object>(
      GeneratedColumnWithTypeConverter<D, S> column) {
    return column.converter.fromSql(read<S>(column));
  }
}
```

Then the query can simply be

```dart
return query.map((e) => CustomModel(e.readWithConverter(mytable.mycolumn)).get();
```

**Reply by Adam-Langley** [reply]
Hi Simon,

Thank you, yes that's correct, its just during select only, joins etc. 

Thanks for the quick response. Glad to know I wasn't missing something obvious 😄 

For your implementation, could I suggest you call the extension function "readValue" instead of "readWithConverter"?
Just for consistency with the query api, where "equalsValue" is the "converter version" of "equals". 

Thoughts?

Thanks again
